apis/ec2/v1beta1/natgateway_types.go

```diff
@@ -0,0 +1,126 @@
+package v1beta1
+
+import (
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Defines the states of NatGateway
+const (
+	NatGatewayStatusPending   = "pending"
+	NatGatewayStatusFailed    = "failed"
+	NatGatewayStatusAvailable = "available"
+	NatGatewayStatusDeleting  = "deleting"
+	NatGatewayStatusDeleted   = "deleted"
+)
+
+// NatGatewayParameters defined the desired state of an AWS VPC NAT Gateway
+type NatGatewayParameters struct {
+
+	// TODO(muvaf): Region is a required field but in order to keep backward compatibility
+	// with old Provider type and not bear the cost of bumping to v1beta2, we're
+	// keeping it optional for now. Reconsider before v1beta2 or v1.
+
+	// Region is the region you'd like your NatGateway to be created in.
+	// +optional
+	// +immutable
+	Region *string `json:"region,omitempty"`
```

Comment:
```suggestion
	// Region is the region you'd like your NatGateway to be created in.
	// +immutable
	Region string `json:"region"`
```

We keep the backward compatibility for v1beta1 resources. So, I think it's safe to make region required.

Reply:
fixed

Reply:
Unresolved this since it looks like it's not changed.

Reply:
`Region` does not have `+optional` set

Reply:
We need to remove `omitempty` and make the type a value instead of pointer, too. See the [patterns doc](https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#pointer-types-and-markers) for details.

Reply:
sorry I missed that, thanks. 👍